### DISCUSSION TITLE
Double manyThrough ?

### DISCUSSION THREAD

**Discussion by Neewd** [adonisjs/core#1890] [Help]
Hi (again),

I have these models :  
Country --> hasMany Regions -> hasMany Locations --> hasMany Pictures 

From there I am able to retrieve the picture with hasManyThrough from the Region model like so 
        return this.manyThrough('App/Models/Location', 'posts')

But now I would like to do the same from the country model like 
        return this.manyThrough('App/Models/Region', 'posts')

But when I try to access this property I have this error 
"self._relatedModelRelation.addWhereOn is not a function", name: "TypeError", status: 500

I have no clue what's wrong :/

**Comment by rrmesquita** [comment]
Have you tried using computed props to solve this? I believe it should work as the following:

```js
'use strict'

class Country extends Model {
  static get computed () {
    return ['countryPosts']
  }

  locations() {
    return this.manyThrough('App/Models/Region', 'locations')
  }

  getCountryPosts () {
    return this.locations.reduce((collection, current) => {
      collection = [...collection, ...current.posts]
      return collection
    }, [])
  }
}
```

**Reply by Neewd** [reply]
Hi @rodrigoriome, thanks for your help. 

Here is now my new model 

```
    static get computed () {
        return ['countryPosts']
    }

    getCountryPosts () {
        return this.locations.reduce((collection, current) => {
          collection = [...collection, ...current.posts]
          return collection
        }, [])
    }

    locations() {
        return this.manyThrough('App/Models/Region', 'locations')
    }

``` 

But when I try to access this property `country.country_posts` or `countryPosts` it's undefined. 
I think this.locations is a function in my case, but when I tried to access it tells me it's not iterable (because it's a many through instance)

**Reply by Neewd** [reply]
I finally make it work, by using simple query and via a serializer where I put back the data with pagination data the way the front want it :) Thank again @rodrigoriome
